samples/tutorial/03_linked_list/sw/linked_list.cpp

```diff
@@ -94,6 +95,10 @@ t_linked_list* initList(t_linked_list* head,
 
         p = p_next;
     }
+
+    // Force all initialization to complete
+    std::atomic<int> mb;
```

Comment:
How does this force initialization to complete?

Reply:
Memory barrier before the CSR write telling the FPGA to read the shared memory. I agree it's unlikely the compiler would every hoist the previous writes, but not technically illegal. Am I missing something?

Reply:
Hmm... I hadn't realized that performing an atomic operations resulted in an `mfence` instruction.
Why not just call  `std::atomic_thread_fence(std::memory_order_seq_cst)`?